sql/log_event.h

```diff
@@ -3404,6 +3412,11 @@ class Gtid_log_event: public Log_event
   static const uchar FL_EXTRA_THREAD_ID= 16; // thread_id like in BEGIN Query
 
 #ifdef MYSQL_SERVER
+  static const uint max_data_length= GTID_HEADER_LEN + 2 + sizeof(XID)
```

Comment:
This new member does not count the size of the user XA's xid which the event embeds with
`FL_*_XA` flags.

Reply:
sizeof(XID) is the one for external XID.